### PR TITLE
Support sygus version 2.1 command assume

### PR DESCRIPTION
**Pull request by ajreynol**
Adds support for global assumptions in sygus files.

Also guards against cases of declare-const, which should be prohibited in sygus.

### CODE REVIEW

File: src/smt/sygus_solver.h [hunk 1]
```diff
@@ -82,8 +82,8 @@ class SygusSolver
                        bool isInv,
                        const std::vector<Node>& vars);
 
-  /** Add a regular sygus constraint.*/
-  void assertSygusConstraint(Node constraint);
+  /** Add a regular sygus constraint or assume.*/
```
Comment (HanielB) [inline src/smt/sygus_solver.h:85]:
```suggestion
  /** Add a regular sygus constraint or assumption.*/
```

File: src/smt/sygus_solver.h [hunk 2]
```diff
@@ -185,6 +185,8 @@ class SygusSolver
   std::vector<Node> d_sygusVars;
   /** sygus constraints */
   std::vector<Node> d_sygusConstraints;
+  /** sygus assumes */
```
Comment (HanielB) [inline src/smt/sygus_solver.h:188]:
```suggestion
  /** sygus assumptions */
```

File: src/parser/smt2/Smt2.g [hunk 3]
```diff
@@ -568,14 +569,13 @@ sygusCommand returns [std::unique_ptr<cvc5::Command> cmd]
           new SynthFunCommand(name, fun, sygusVars, range, isInv, grammar));
     }
   | /* constraint */
-    CONSTRAINT_TOK {
+    ( CONSTRAINT_TOK { isAssume = false; } | ASSUME_TOK { isAssume = true; } )
```
Comment (HanielB) [inline src/parser/smt2/Smt2.g:572]: kill trailing space

**Review by HanielB — APPROVED** [review]
LGTM, just minor changes

**Review by abdoo8080 — APPROVED** [review]

**Pull request merged**